### ISSUE TITLE
key of 2022-blake3-aes-128gcm with16-byte key can't work

### ISSUE THREAD

**Issue by aria001** [SagerNet/SagerNet#461]
**Before submitting an issue, please make sure you are using the latest version of SagerNet.**

### Describe the bug

key of 2022-blake3-aes-128-gcm with16-byte key can't work

### To Reproduce

1. Make a configuration of shadowsocks 2022-blake3-aes-128-gcm with16-byte key
2. Turn on the airplane icon
3. Gots error  "fail:: bad key"

### Debug Information

> About -> Export Debug Information

**If you do not send the file directly, please use the Markdown code block format, otherwise your
issue will be closed!**

```
05-12 12:28:38.398   851   883 D v2ray-core: app/log: Logger started
05-12 12:28:38.413   851  1010 D v2ray-core: app/proxyman/inbound: creating stream worker on 0.0.0.0:2080
05-12 12:28:38.413   851  1010 D v2ray-core: app/proxyman/inbound: creating stream worker on 0.0.0.0:9080
05-12 12:28:38.413   851  1010 D v2ray-core: app/proxyman/inbound: creating stream worker on 0.0.0.0:6450
05-12 12:28:38.413   851  1010 D v2ray-core: app/stats: create new counter outbound>>>proxy-global-55>>>traffic>>>uplink
05-12 12:28:38.413   851  1010 D v2ray-core: app/stats: create new counter outbound>>>proxy-global-55>>>traffic>>>downlink
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: go.Universe$proxyerror: bad key
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at libcore.V2RayInstance.loadConfig(Native Method)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.proto.V2RayInstance.loadConfig(V2RayInstance.kt:1)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.proto.V2RayInstance.init(V2RayInstance.kt:60)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.proto.ProxyInstance.init(ProxyInstance.kt:1)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.BaseService$Interface$onStartCommand$2.invokeSuspend(BaseService.kt:6)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:24)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:1)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at kotlinx.coroutines.AbstractCoroutine.start$enumunboxing$(AbstractCoroutine.kt:12)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:6)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.ktx.AsyncsKt.runOnMainDispatcher(Asyncs.kt:3)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.BaseService$Interface$DefaultImpls.onStartCommand(BaseService.kt:21)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at io.nekohasekai.sagernet.bg.VpnService.onStartCommand(VpnService.kt:8)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4442)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.app.ActivityThread.access$2000(ActivityThread.java:252)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2037)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.os.Handler.dispatchMessage(Handler.java:106)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.os.Looper.loop(Looper.java:250)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at android.app.ActivityThread.main(ActivityThread.java:7886)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at java.lang.reflect.Method.invoke(Native Method)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
05-12 12:28:38.413   851   851 W BaseService$Interface$onStartCommand$2: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
```

### Expected behavior

2022-blake3-aes-128-gcm should works with a 16-byte PSK

### Smartphone (please complete the following information)

- Kernel version: [e.g. 4.19.114]
- Device: [e.g. MI6]
- OS: [e.g. Android8.1]

**Comment by aria001** [comment]
2022-blake3-aes-256-gcm with 32-byte key works well.

**Comment by nekohasekai** [comment]
All shadowsocks 2022 ciphers use 32 bit key.

**Comment by aria001** [comment]
Then it should be a mistake of shadowscosks/shadowsocks-rust's implemention, sorry.

**Comment by nekohasekai** [comment]
Looks like my problem, the spec has changed to 16b

**Comment by aria001** [comment]
Fixes in 0.8-beta04 make switch can be turned on but handshake wents failed. 
Seems there's not any useful infomation in client's log, log in server (shadowsocks-rust)only shows:
```
handshake failed, maybe wrong method or key, or under replay attacks. peer: [::ffff:xxx.xxx.xxx.xxx]:42096, error: invalid tag-in
```
Should I change server to Sing for further test?

**Comment by nekohasekai** [comment]
Shadowsocks rust implementation may have more differences, please wait for negotiation. Until then, use sing ss-server.
